packages/engine/lib/stateful/src/field/spec_map.rs

```diff
@@ -152,13 +153,16 @@ impl<S: FieldSource + Clone> FieldSpecMap<S> {
 
         let mut metadata = HashMap::with_capacity(1);
         // TODO: this can be simplified when we update arrow-rs (beyond 1.0.1), we can set this on
-        // Field's custom metadata instead of the schema
+        //   Field's custom metadata instead of the schema
         metadata.insert("any_type_fields".into(), any_types.join(","));
         metadata.insert("nullable".into(), nullabilities.join(","));
         Ok(Schema::new_with_metadata(
             partitioned_fields
                 .into_iter()
                 .map(|(field_spec, _)| Field::try_from(field_spec.clone()))
+                // TODO: Can this ever fail? `Field::try_from` fails, if it has an invalid 
+                //   `FieldKey`. Possible solution: Use `name: FieldKey` in `FieldSpec` rather than 
```

Comment:
Can't do that because FieldKeys are only valid for RootFieldSpecs, they are FieldSpecs do not have a key. (e.g. a nested field inside a struct)

Reply:
That's a left-over TODO I added before we talked about this off-Github. I removed it in 71cef3b29e4b3ed1e7bed45c54f9891c7e398b34